ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,210 @@
+  - [2.2 Scenario: Show available paste formats in web based editors](#22-scenario-show-available-paste-formats-in-web-based-editors)
+    - [2.2.1 Copy multiple cells in a spreadsheet should show multiple paste options](#221-copy-multiple-cells-in-a-spreadsheet-should-show-multiple-paste-options)
+    - [2.2.2 Copy plain text should show only single paste option](#222-copy-plain-text-should-show-only-single-paste-option)
+- [3. Motivation - Alternative to inefficient polling of clipboard](#3-motivation---alternative-to-inefficient-polling-of-clipboard)
+- [4. Example javascript code for detecting clipboard changes:](#4-example-javascript-code-for-detecting-clipboard-changes)
+- [5. Event spec details and open questions](#5-event-spec-details-and-open-questions)
+  - [5.1 User permission requirement](#51-user-permission-requirement)
+    - [5.1.1 Approach 1 - clipboard-read permission required to listen to clipboardchange event](#511-approach-1---clipboard-read-permission-required-to-listen-to-clipboardchange-event)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [5.1.2 Approach 2 - No permission required](#512-approach-2---no-permission-required)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+    - [5.1.3 Conclusion](#513-conclusion)
+  - [5.2 Page focus requirement](#52-page-focus-requirement)
+    - [5.2.1 Approach 1 - Page required to be in focus to receive event](#521-approach-1---page-required-to-be-in-focus-to-receive-event)
+      - [Pros](#pros-2)
+      - [Cons](#cons-2)
+    - [5.2.2 Approach 2 - No focus requirement](#522-approach-2---no-focus-requirement)
+      - [Pros:](#pros)
+      - [Cons:](#cons)
+    - [5.2.3 Conclusion](#523-conclusion)
+  - [5.3 Event details](#53-event-details)
+- [6. Detailed design discussion](#6-detailed-design-discussion)
+  - [6.1 Listen to clipboard change directly from the OS](#61-listen-to-clipboard-change-directly-from-the-os)
+      - [Pros:](#pros-1)
+      - [Cons:](#cons-1)
+    - [6.1.1 APIs provided by all OS to listen to clipboardchange event:](#611-apis-provided-by-all-os-to-listen-to-clipboardchange-event)
+  - [6.2 Considered alternative: Given the page focus restrictions, check clipboard hash change when page regains focus](#62-considered-alternative-given-the-page-focus-restrictions-check-clipboard-hash-change-when-page-regains-focus)
+    - [6.2.1 Platform support for getting clipboard SequenceNumber/Version number](#621-platform-support-for-getting-clipboard-sequencenumberversion-number)
+      - [Pros:](#pros-2)
+      - [Cons:](#cons-2)
+- [7. References & acknowledgements](#7-references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## 1. Introduction
+
+The clipboardchange event fires whenever the system clipboard contents are changed. This allows web-apps like remote desktop clients to stay in sync with the system clipboard. It provides an efficient alternative to polling the clipboard for changes.
```

Comment:
"This allows web-apps like remote desktop clients to stay in sync with the system clipboard." -> "Staying in sync" implies reading the clibpoard, which this API doesn't provide the ability to do. Maybe we should say "allows web apps to be notified and respond to changes to the system clipboard".